query/src/main/scala/filodb/query/exec/RangeVectorTransformer.scala

```diff
@@ -111,36 +137,44 @@ final case class InstantVectorFunctionMapper(function: InstantFunctionId,
 
 private class DoubleInstantFuncIterator(rows: Iterator[RowReader],
                                         instantFunction: DoubleInstantFunction,
+                                        scalar: Seq[ScalarRangeVector],
                                         result: TransientRow = new TransientRow()) extends Iterator[RowReader] {
   final def hasNext: Boolean = rows.hasNext
   final def next(): RowReader = {
     val next = rows.next()
-    val newValue = instantFunction(next.getDouble(1))
-    result.setValues(next.getLong(0), newValue)
+    val nextVal = next.getDouble(1)
+    val timestamp = next.getLong(0)
+    val newValue = instantFunction(nextVal, scalar.map(_.getValue(timestamp)))
+    result.setValues(timestamp, newValue)
     result
   }
 }
 
 private class H2DoubleInstantFuncIterator(rows: Iterator[RowReader],
                                           instantFunction: HistToDoubleIFunction,
+                                          scalar: Seq[ScalarRangeVector],
                                           result: TransientRow = new TransientRow()) extends Iterator[RowReader] {
   final def hasNext: Boolean = rows.hasNext
   final def next(): RowReader = {
     val next = rows.next()
-    val newValue = instantFunction(next.getHistogram(1))
-    result.setValues(next.getLong(0), newValue)
+    val timestamp = next.getLong(0)
+    val newValue = instantFunction(next.getHistogram(1), scalar.map(_.getValue(timestamp)))
```

Comment:
This seems potentially really expensive - how big is the scalar?  map produces a new Seq for every sample....  can you please explain what is going on here?   Since most of the time there is no scalar, I wonder if we can make this a NOP most of the time.

Reply:
Yes that is because the scalar can be an ExecPlan which will have a different value for every timestamp.